src/_guides/libraries/private-files.md

```diff
@@ -103,7 +103,7 @@ similar to Ruby's `Gemfile.lock`.
 Regenerating the `pubspec.lock` file lets you test your package
 against the latest compatible versions of its dependencies.
 
-**For application packages**, we recommend that you
-**commit** the `pubspec.lock` file.
-Saving `pubspec.lock` ensures that everyone working on the app
-uses the exact same versions.
+**For application packages**, we recommend that you commit
+the `pubspec.lock` file.
+Saving `pubspec.lock` will track the dependencies in production or
+development, it can help you locate problems due to dependency changes.
```

Comment:
Is your concern that the previous statement accidentally suggested `pubspec.lock` won't ever be affected by `pub get` and we just need to be more explicit about the benefits of checking in the lockfile?

How does something like the following sound?

```suggestion
**For application packages**, we recommend that you
**commit** the `pubspec.lock` file.
Saving `pubspec.lock` ensures dependency updates are explicit
and don't unexpectedly change during the development process
or when working with other developers.
```

Reply:
I never understood the `.lock` file to begin with, so bear with me.

Is the idea that for an *application* package, it's important to be able to recreate the the released code.
That means knowing the precise versions of all the packages that are included in the final program, and to that end, you can commit the `pubspec.lock` file used to build the release version as part of the versioning of your development environment.

If so, you don't need to commit `pubspec.lock` on your development branch, but can, if you want to make sure you have a stable environment for everybody working on the code. 

You can get the same effect by precisely versioning your dependencies in the pubspec.yaml, so `test: 1.16.2` instead of a allowing a range like `test: ^1.10`. The advantage of using the `.lock` file is that you can use tools to update it when you want it to be updated, where you have to manually modify `pubspec.yaml`.

Am I understanding it correctly?

Reply:
> You can get the same effect by precisely versioning your dependencies in the pubspec.yaml,

You'd need to use `dependency_overrides` to pin a version for transitive dependencies. This is not a pattern we recommend.

I like the framing that using version control makes changes to dependencies explicit.

"Using version control on `pubspec.lock` makes changes in transitive dependencies explicit. Each time the dependencies change from a `dart pub upgrade` or a change in `pubspec.yaml` the difference will apparent in the lock file."

I don't know if it's important for us to say why an author may want dependencies to change or not change - that feels outside the scope here.

Reply:
Sounds good to me, thanks! Something like the following:

```suggestion
**For application packages**, 
we recommend that you commit the `pubspec.lock` file.
Versioning the `pubspeck.lock` file
ensures changes to transitive dependencies are explicit.
Each time the dependencies change due to `dart pub upgrade`
or a change in `pubspec.yaml` 
the difference will be apparent in the lock file.
```

Reply:
Refine the description about the suggestion